bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -477,12 +506,20 @@ void ReplicaImp::onMessage<preprocessor::PreProcessResultMsg>(preprocessor::PreP
 
 template <>
 void ReplicaImp::onMessage<ReplicaAsksToLeaveViewMsg>(ReplicaAsksToLeaveViewMsg *m) {
+  if (activeExecutions_ > 0) {
+    deferredMessages_.push_back(m);
+    deferredMessagesMetric_++;
+    return;
+  }
   MDC_PUT(MDC_SEQ_NUM_KEY, std::to_string(getCurrentView()));
   if (m->viewNumber() == getCurrentView()) {
     LOG_INFO(VC_LOG,
              "Received ReplicaAsksToLeaveViewMsg " << KVLOG(m->viewNumber(), m->senderId(), m->idOfGeneratedReplica()));
     viewsManager->storeComplaint(std::unique_ptr<ReplicaAsksToLeaveViewMsg>(m));
-    tryToGotoNextView();
+    if (activeExecutions_ > 0)
```

Comment:
same

Reply:
When activeExecutions_ > 0, it will return from Line#521, in that case will this line will be executed?

Reply:
Amitava is right, removed this if statement.